webapp/src/components/ImageEditor/toolDefinitions.ts

```diff
@@ -659,6 +666,20 @@ export class FillEdit extends Edit {
 }
 
 
+export class WallFillEdit extends FillEdit {
```

Comment:
Instead of creating a new tool, I think you could just replace the existing fill behavior with this. Always fill the active layer but base it off of the tile layer; should work for both cases!

You might need to expose the activeLayer on the tilestate to get around the `if (colorToReplace === this.color)` check for walls

Reply:
Nice. Sounds good.

There is a small optimization in Fill Edit's doEditCore that skips the work if the target tile is the same as the color you're filling. This doesn't make sense when you're operating on different layers, but I'm sure I can figure something reasonable out.